### ISSUE TITLE
Index page does not load after marketplace has been set up.

### ISSUE THREAD

**Issue by prinzdezibel** [reactioncommerce/reaction#3846]
## Prerequisites
* [x] Are you running the latest version?
* [x] Are you able to consistently reproduce the issue?
* [x] Did you search the issue queue for existing issue? [Search issues](https://github.com/reactioncommerce/reaction/issues)

## Issue Description
After inviting a merchant to join the shop, the home page will not render anymore, because the spinner is loading forever and an error can be seen in console:
```
TypeError: Cannot read property 'name' of undefined
    at TagNav.renderShopSelect (app.js:77416)
    at TagNav.render (app.js:77445)
    at modules.js:154734
    at measureLifeCyclePerf (modules.js:154014)
    at ReactCompositeComponentWrapper._renderValidatedComponentWithoutOwnerOrContext (modules.js:154733)
    at ReactCompositeComponentWrapper._renderValidatedComponent (modules.js:154760)
    at ReactCompositeComponentWrapper.performInitialMount (modules.js:154300)
    at ReactCompositeComponentWrapper.mountComponent (modules.js:154196)
    at Object.mountComponent (modules.js:147051)
    at ReactDOMComponent.mountChildren (modules.js:153322)
```

## Steps to Reproduce
Please provide starting context, i.e. logged in as a user, configure a particular payment method.
1. As a shop admin, set up marketplace and invite another user
2. Sign out as admin
3. Close tab.
4. Check emails of invited person and enroll account.
5. Set password for new account
6. Sign out
7. Refresh page 
8. See spinner is loading endlessly and the console shows the above error.

## Possible Solution
It has to do with the local storage. Once you delete the **local storage** for the domain:port in question, it works again.

### Versions
Node: 8.2.1
NPM: 5.3.0
Meteor Node: 8.9.4
Meteor NPM: 5.6.0
Reaction CLI: 0.26.4
Reaction: 1.8.1
Reaction branch: master
Docker: 17.03.1-ce

**Comment by zenweasel** [comment]
I think this only impacts you if you were previously signed in as the shop admin and then logout (so that this inactive shop is stored in local storage as your active shop, but that shop is not publicly available). So this issue should never impact a consumer-facing use case so it's not really a critical.

**Comment by zenweasel** [comment]
Ok, so this does affect you if you login as a shop admin and then as a marketplace admin all sorts of weird stuff happens because your `activeShopId` is set to the last shop, and there are probably other side-effects that I am not seeing so bumping it back up to critical fwiw.

**Comment by mikemurray** [comment]
I'm seeing the following error in the accounts panel as the owner of the merchant shop looking at account settings. Sometimes there's no error and it's just a broken accounts panel. Not sure if this is somehow related, or a different issue altogether.

```
TypeError: Cannot read property 'permissions' of undefined
    at PermissionsList._this.checked (permissionsList.js:70)
    at permissionsList.js:103
    at Array.forEach (<anonymous>)
    at PermissionsList.renderPermissions (permissionsList.js:98)
    at PermissionsList.render (permissionsList.js:116)
    at finishClassComponent (modules.js?hash=469040b828aef9a0170c6dc8d06d0783091cd078:148404)
    at updateClassComponent (modules.js?hash=469040b828aef9a0170c6dc8d06d0783091cd078:148381)
    at beginWork (modules.js?hash=469040b828aef9a0170c6dc8d06d0783091cd078:148756)
    at performUnitOfWork (modules.js?hash=469040b828aef9a0170c6dc8d06d0783091cd078:150755)
    at workLoop (modules.js?hash=469040b828aef9a0170c6dc8d06d0783091cd078:150819)
```

![screen shot 2018-03-02 at 4 20 56 pm](https://user-images.githubusercontent.com/42217/36927902-b9d6fd5c-1e35-11e8-840e-0b25a02a6aa7.png)

**Comment by zenweasel** [comment]
Did you try this with the patch for it @mikemurray ?

**Comment by mikemurray** [comment]
@zenweasel seems the patch fixes this issue as well

**Comment by mikemurray** [comment]
After some additional testing... 

It would seem that the active `shopId` is wrong on the server and correct on the client. 

- Server `Reaction.getShopId() === "J8Bhq3uTtdgwZx3rz"` in `ShopMembers` publication.
- Client `Reaction.getShopId() === "Wjp8rCDf6z8HeA3mK"` where `ShopMembers` is subscribed at.

@zenweasel This patch does not totally fix the issue, it must have bee a fluke that it worked the time I tested it.

@prinzdezibel I think the issue might be that LocalStorage is getting the proper active shop, but that change is not being reflected on the server, so in places where we rely on the server `Reaction.getShopId()` fail for marketplace shops.

PR https://github.com/reactioncommerce/reaction/pull/3463 may provide some clues to help resolve this issue.

@Akarshit thoughts on PR #3463 and how we handle the active shopId on the server?

**Comment by zenweasel** [comment]
@mikemurray Can you provide some reproduction steps for this? Should we open a new/separate issue?

**Comment by mikemurray** [comment]
@zenweasel I feel like it may be the same issue, perhaps related to `shopId` not being accurately set for marketplace shops between client/server.

1. Sign in as admin
1. Enable marketplace user Shop settings
1. Invite new shop owner
1. Make shop active under `Marketplace Shops` in the sidebar
1. Accept invitation
1. Set password and continue. (You should now be signed in as the merchant shop owner)
1. Open the accounts panel
1. See no accounts, or groups are available to edit
1. With no permissions loaded, you cant even create products or update any settings

**Comment by zenweasel** [comment]
@mikemurray Not able to replicate the problem using the test steps above:

![2018-03-06_08-08-59](https://user-images.githubusercontent.com/72819/37007125-c87f9826-2116-11e8-9bc7-cf119cbf54b9.png)

**Comment by Akarshit** [comment]
Same (not able to replicate the issue)

**Comment by mikemurray** [comment]
@zenweasel @Akarshit with release `release-1.9.0` branch I'm not seeing the issue for the merchant shop owner, but I am seeing it as the marketplace owner.

1. As the marketplace owner, after inviting a marketplace shop
1. Use the shop switcher to switch to that shop
1. Open accounts panel
1. See blank/broken accounts view

---

My issue is happening with PR https://github.com/reactioncommerce/reaction/pull/3876 (faster product grid) which is branched off of https://github.com/reactioncommerce/reaction/pull/3782 (replace CFS); I'll have to see if an update from the `release-1.9.0` branch resolves the issue.

**Comment by Akarshit** [comment]
Though I was looking at the `activeShopId` in the localStorage while doing this and observed the following.
1. Sign in as admin(for Reaction)
1. Enable marketplace user Shop settings
1. Invite new shop(Reaction1) owner
1. Make shop active under Marketplace Shops in the sidebar
1. Accept invitation
1. Set password and continue.
1. When you log-in as owner of Reaction1 the `activeShopId` in LS is `Reaction1`
1. Log out (`activeShopId` is `Reaction1`)
1. Log in as admin for `Reaction`, `activeShopId` is still `Reaction1`

This could be fixed repeating #3856 [here](https://github.com/reactioncommerce/reaction/blob/master/client/modules/core/main.js#L375)

After doing the above the process changes to
1. When you log-in as owner of Reaction1 the `activeShopId` in LS is `Reaction1`
1. Log out (`activeShopId` is `Reaction`)
1. Log in as admin for `Reaction`, `activeShopId` is still `Reaction`
Which is correct AFAIK

**Comment by Akarshit** [comment]
@mikemurray Yes I can see it happening with marketplace owner. Looking more into it.

**Comment by Akarshit** [comment]
This is happening because the Groups aren't being published when the shopId changes. So just changing `const grpSub = Meteor.subscribe("Groups")`; to `const grpSub = Meteor.subscribe("Groups", { shopId });` [here](https://github.com/reactioncommerce/reaction/blob/master/imports/plugins/core/accounts/client/containers/accountsDashboardContainer.js#L94) fixes this.

The Groups are not being published because the code inside the publication on server-side is not reactive.

**Comment by Akarshit** [comment]
Ya sure.

**Comment by zenweasel** [comment]
So this issue was resolved by #3856. @akarshit has opened a new ticket for the issue that @mikemurray noted before.

#3942
